common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -597,6 +598,30 @@ public static CharSequence trimOws(CharSequence value) {
         return start == 0 && end == length - 1 ? value : value.subSequence(start, end + 1);
     }
 
+    /**
+     * Returns a new string that contains all {@code elements} joined by a given separator.
+     *
+     * @param separator for each element
+     * @param elements to join together
+     *
+     * @return a string joined by a given separator.
+     */
+    public static CharSequence join(CharSequence separator, Iterator<? extends CharSequence> elements) {
+        ObjectUtil.checkNotNull(separator, "separator");
+        ObjectUtil.checkNotNull(elements, "elements");
+
+        if (!elements.hasNext()) {
+            return EMPTY_STRING;
+        }
+
+        StringBuilder builder = new StringBuilder().append(elements.next());
```

Comment:
I wonder if we should optimise this a but to only build the StringBuilder etc when there are at least 2 elements in ?

Reply:
This could also be simplified to just `new StringBuilder(elements.next())`.

Reply:
All done :)